code/go/internal/validator/semantic/validate_duration_variables.go

```diff
@@ -0,0 +1,288 @@
+}
+
+// allVars collects all duration variables from a manifest including those at the
+// top level, within policy templates, and nested inside inputs. It returns a
+// flattened slice of all variables for easier processing.
+func (m *durationManifest) allVars() []durationVar {
+	var out []durationVar
+	out = append(out, m.Vars...)
+	for _, t := range m.PolicyTemplates {
+		out = append(out, t.Vars...)
+		for _, i := range t.Inputs {
+			out = append(out, i.Vars...)
+		}
+	}
+	return out
+}
+
+// durationDataStreamManifest represents the structure of a data stream manifest
+// file focusing on the stream elements that may contain duration variables.
+type durationDataStreamManifest struct {
+	Streams []struct {
+		Vars []durationVar `yaml:"vars"` // Stream variables
+	} `yaml:"streams"`
+}
+
+// allVars collects all duration variables from a data stream manifest. It
+// returns a flattened slice of all variables from all streams for easier
+// processing.
+func (m *durationDataStreamManifest) allVars() []durationVar {
+	var out []durationVar
+	for _, s := range m.Streams {
+		out = append(out, s.Vars...)
+	}
+	return out
+}
+
+// validateDurationVar checks that duration variables follow the required constraints:
+//
+// 0 <= min_duration <= default <= max_duration, for variables of type "duration".
+//
+// It parses the duration strings, validates their format, and ensures they meet
+// the ordering constraints. Multiple validation errors may be returned together
+// using errors.Join if the variable has multiple constraint violations.
+func validateDurationVar(v durationVar) error {
+	// Only validate variables of the type "duration".
+	if v.Type != "duration" {
+		return nil
+	}
+
+	var minDuration, defaultDuration, maxDuration time.Duration
+	var err error
+
+	// Parse min_duration if defined
+	if v.MinDuration != nil {
+		minDuration, err = time.ParseDuration(*v.MinDuration)
+		if err != nil {
+			return fmt.Errorf("invalid min_duration value %q: %w", *v.MinDuration, err)
+		}
+		// Ensure min_duration is not negative
+		if minDuration.Nanoseconds() < 0 {
+			return fmt.Errorf("negative min_duration value %q", *v.MinDuration)
+		}
+	}
+
+	// Parse default if defined
+	switch s := v.Default.(type) {
+	case nil:
+	case string:
+		defaultDuration, err = time.ParseDuration(s)
+		if err != nil {
+			return fmt.Errorf("invalid default value %q: %w", s, err)
+		}
+	default:
+		return fmt.Errorf("invalid default value type %T: expected a string", s)
+	}
+
+	// Parse max_duration if defined
+	if v.MaxDuration != nil {
+		maxDuration, err = time.ParseDuration(*v.MaxDuration)
+		if err != nil {
+			return fmt.Errorf("invalid max_duration value %q: %w", *v.MaxDuration, err)
+		}
+	}
+
+	// Check constraints: 0 <= min_duration <= default <= max_duration
+	var errs []error
+
+	// Check min_duration <= default (if both are defined)
+	if v.MinDuration != nil && v.Default != nil {
+		if minDuration.Nanoseconds() > defaultDuration.Nanoseconds() {
```

Comment:
Durations can be directly compared, right?
```suggestion
		if minDuration > defaultDuration {
```

Reply:
Absolutely 👍 . bda8b4c